test/e2e/cluster_image_registry_operator_test.go

```diff
@@ -2,25 +2,83 @@ package e2e_test
 
 import (
 	"testing"
+	"time"
 
-	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+
+	coreapi "k8s.io/api/core/v1"
+	metaapi "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorapi "github.com/openshift/api/operator/v1alpha1"
+
+	imageregistryapi "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1alpha1"
 )
 
 func TestClusterImageRegistryOperator(t *testing.T) {
-	ctx := framework.NewTestCtx(t)
-	defer ctx.Cleanup(t)
+	namespace := Framework.GetNamespace(t)
+	defer namespace.Cleanup()
 
-	err := ctx.InitializeClusterResources()
-	if err != nil {
-		t.Fatalf("failed to initialize cluster resources: %v", err)
+	cr := &imageregistryapi.ImageRegistry{
+		TypeMeta: metaapi.TypeMeta{
+			Kind:       "ImageRegistry",
+			APIVersion: imageregistryapi.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metaapi.ObjectMeta{
+			Name:      "image-registry",
+			Namespace: namespace.Name,
+		},
+		Spec: imageregistryapi.ImageRegistrySpec{
+			OperatorSpec: operatorapi.OperatorSpec{
+				ManagementState: "Managed",
+			},
+			Storage: imageregistryapi.ImageRegistryConfigStorage{
+				Filesystem: &imageregistryapi.ImageRegistryConfigStorageFilesystem{
+					VolumeSource: coreapi.VolumeSource{
+						EmptyDir: &coreapi.EmptyDirVolumeSource{},
+					},
+				},
+			},
+			Replicas: 1,
+		},
 	}
+	if err := sdk.Create(cr); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		// TODO(dmage): can we do it better?
```

Comment:
any reason not to just delete the whole namespace?

Reply:
The whole namespace will be deleted by the deferred Cleanup call. But first we need to remove the finalizer from our CR to avoid the deadlock that is described at #38.

Reply:
I forgot that wasn't fixed yet, but let's go ahead and fix it rather than working around it in the tests, it should be an easy change.